tests_e2e/tests/ext_cgroups/install_extensions.py

```diff
@@ -0,0 +1,139 @@
+        # As part of pilot testing , agent hardcoded azuremonitoragent service name to monitor it for sometime in production without need of manifest update from extesnion side.
+        # So that they can get sense of resource usage for their extensions. This we did for few months and now we no logner monitoring it in production.
+        # But I'm mocking the same behaviour here in test by changing the expiry time to future date. So that test agent will start track the cgroups that is used by the service.
+        result = self._ssh_client.run_command(f"update-waagent-conf Debug.CgroupMonitorExpiryTime={expiry_time}", use_sudo=True)
+        log.info(result)
+        log.info("=====Updated agent cgroups config(CgroupMonitorExpiryTime)=====")
+
+    def _install_ama(self):
+        ama_extension = VirtualMachineExtensionClient(
+            self._context.vm, VmExtensionIds.AzureMonitorLinuxAgent,
+            resource_name="AMAAgent")
+        log.info("Installing %s", ama_extension)
+        ama_extension.enable()
+        ama_extension.assert_instance_view()
+
+    def _install_vmaccess(self):
+        # fetch the public key
+        public_key_file: Path = Path(self._context.private_key_file).with_suffix(".pub")
+        with public_key_file.open() as f:
+            public_key = f.read()
+        # Invoke the extension
+        vm_access = VirtualMachineExtensionClient(self._context.vm, VmExtensionIds.VmAccess, resource_name="VmAccess")
+        log.info("Installing %s", vm_access)
+        vm_access.enable(
+            protected_settings={
+                'username': self._context.username,
+                'ssh_key': public_key,
+                'reset_ssh': 'false'
+            }
+        )
+        vm_access.assert_instance_view()
+
+    def _install_gatest_extension(self):
+        gatest_extension = VirtualMachineExtensionClient(
+            self._context.vm, VmExtensionIds.GATestExtension,
+            resource_name="GATestExt")
+        log.info("Installing %s", gatest_extension)
+        gatest_extension.enable()
+        gatest_extension.assert_instance_view()
+
+
+    def _install_cse(self):
+        # Use custom script to output the cgroups assigned to it at runtime and save to /var/lib/waagent/tmp/custom_script_check.
+        # Output the status of the agent when both CSE and AMA are running to make sure none of them were assigned
+        # to the agent by mistake; save to /var/lib/waagent/tmp/walinuxagent_check_after_ama.
+        script_contents = """#!/usr/bin/env bash
+function check_ama {
+    while true;
+    do
+        ama_dir=$(find /var/lib/waagent -maxdepth 1 -type d -name "Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-*" -print -quit)
+        grep -i "Enable succeeded" $ama_dir/status/0.status &> /dev/null
+        if [ $? -eq 0 ];
+        then
+            ps aux --forest > /var/lib/waagent/tmp/ps_check_after_ama
```

Comment:
I'm begging to think why we need process info at all. what extra information it's providing? and also checking enable succeeded doesn't make sense since we check the provision status when we install extension